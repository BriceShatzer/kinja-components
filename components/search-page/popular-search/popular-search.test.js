/* @flow */

import * as React from 'react';
import { mount } from 'enzyme';

import PopularSearch, { Keyword, Title } from './popular-search';


describe('<PopularSearch />', () => {
	const onClick = jest.fn();
	const stubElement = ({ blogName = 'Jalopnik' } = {}) => (
		<PopularSearch
			blogName={blogName}
			keywords={['Uruk-hai', 'Goblins', 'Moria Orcs', 'Black Uruks']}
			onClick={onClick}
			theme="jalopnik"
		/>
	);
	it('should render and match snapshot', () => {
		const popularSearch = mount(stubElement());
		expect(popularSearch).toMatchSnapshot();
	});

	it('should add blogName to the title', () => {
		const blogName = 'Shire';
		const popularSearch = mount(stubElement({ blogName }));

		const title = popularSearch.find(Title).text();
		expect(title).toBe(`Popular searches on ${blogName}`);
	});

	it('should call onClick', () => {
		const popularSearch = mount(stubElement());
		const keyword = popularSearch.find(Keyword).at(1);

		keyword.simulate('click');
		expect(onClick).toHaveBeenCalledTimes(1);
		expect(onClick).toHaveBeenCalledWith('Goblins');
	});
});
