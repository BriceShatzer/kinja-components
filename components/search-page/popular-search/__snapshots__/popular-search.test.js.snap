// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopularSearch /> should render and match snapshot 1`] = `
.c2 {
  margin-bottom: 10px;
}

.c5 {
  margin-bottom: 4px;
  color: #3E79B1;
  cursor: pointer;
}

.c5:hover {
  color: #315f8b;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
}

.c0 .c1,
.c0 .c4 {
  font-weight: bold;
  font-size: 15px;
  line-height: 19px;
}

@media only screen and (min-width:50em) {
  .c0 {
    padding: 8px 10px 10px;
    border: 1px solid #3E79B1;
  }

  .c0 .c1 {
    margin-bottom: 14px;
    font-size: 18px;
    line-height: 22px;
  }

  .c0 .c4 {
    font-size: 20px;
    line-height: 24px;
    margin-bottom: 8px;
  }
}

<PopularSearch
  blogName="Jalopnik"
  keywords={
    Array [
      "Uruk-hai",
      "Goblins",
      "Moria Orcs",
      "Black Uruks",
    ]
  }
  locale="en-US"
  onClick={[MockFunction]}
  theme="jalopnik"
>
  <Theme
    blog="jalopnik"
    theme={
      Object {
        "breakpointBanner": "969px",
        "breakpointMedium": "53.125em",
        "color": Object {
          "alert": "#EEB544",
          "amazon": "#F5C056",
          "backgroundLayer": "rgba(0,0,0,0.45)",
          "backgroundLight": "#E6F2F8",
          "black": "#000",
          "blackOverlay": "rgba(0,0,0,0.7)",
          "bodytext": "#222",
          "commerce": "#72AD75",
          "darkBlackOverlay": "rgba(0,0,0,0.9)",
          "darkgray": "#666",
          "darksmoke": "#222",
          "error": "#D24A1D",
          "facebook": "#1877F2",
          "foregroundLayer": "rgba(0,0,0,0.4)",
          "google": "#DB4437",
          "gray": "#7D7D7D",
          "instagram": "#C32AA3",
          "lightgray": "#E5E5E5",
          "linkedin": "#0077B5",
          "logo": "#000",
          "midgray": "#CCC",
          "primary": "#0A7BC2",
          "primaryLight": "#0C91E4",
          "rss": "#7D7D7D",
          "secondarytext": "#7D7D7D",
          "success": "#23891F",
          "twitter": "#1DA1F2",
          "white": "#FFF",
          "whiteOverlay": "rgba(255,255,255,0.7)",
          "whitesmoke": "#F5F5F5",
          "youtube": "#FF0000",
        },
        "columnGutter": "2.25rem",
        "columnGutter19": Object {
          "largeUp": "1.5rem",
          "small19": "1rem",
          "xlargeUp": "2rem",
        },
        "columnPadding": "1.125rem",
        "columnPadding19": Object {
          "largeUp": "0.75rem",
          "small19": "0.5rem",
          "xlargeUp": "1rem",
        },
        "featuredContentWidth": "1024px",
        "globalNavHeight": "80px",
        "linkTransition": "0.1s ease-out",
        "mainContainerWidth": "66.5%",
        "mainContentMaxWidth": "800px",
        "pageWidth": "1336px",
        "postContentMaxWidth": "636px",
        "scrollbackNavHeight": "60px",
        "sidebarContainerWidth": "33.5%",
        "sidebarContentMaxWidth": "360px",
        "sidebarContentMinWidth": "300px",
        "slideshowTransitionDuration": "0.2s",
        "typography": Object {
          "body": Object {
            "fontFamily": "ElizabethSerif, Georgia, serif",
            "fontSizes": Object {
              "large": "27px",
              "medium": "19px",
              "small": "15px",
              "xlarge": "30px",
              "xsmall": "14px",
            },
            "lineHeights": Object {
              "large": "40px",
              "medium": "32px",
              "small": "24px",
              "xlarge": "44px",
              "xsmall": "24px",
            },
          },
          "embiggenedHeadlineSize": "34px",
          "headline": Object {
            "fontFamily": "ProximaNovaCond, sans-serif",
            "fontSizes": Object {
              "large": "32px",
              "medium": "24px",
              "small": "19px",
              "xlarge": "37px",
              "xsmall": "16px",
            },
            "lineHeights": Object {
              "large": "40px",
              "medium": "32px",
              "small": "24px",
              "xlarge": "44px",
              "xsmall": "24px",
            },
          },
          "headlineLineHeight": "1.3",
          "headlineSize": "24px",
          "permalinkHeadlineLineHeight": "1.2",
          "permalinkHeadlineSize": Object {
            "mediumUp": "2.5rem",
            "small": "2rem",
            "xxxlargeUp": "3rem",
          },
          "primary": Object {
            "fontFamily": "ProximaNovaCond, sans-serif",
            "fontSizes": Object {
              "large": "32px",
              "medium": "24px",
              "small": "19px",
              "xlarge": "37px",
              "xsmall": "16px",
            },
            "lineHeights": Object {
              "large": "40px",
              "medium": "32px",
              "small": "24px",
              "xlarge": "44px",
              "xsmall": "24px",
            },
          },
          "serif": Object {
            "fontFamily": "ElizabethSerif, Georgia, serif",
            "fontSizes": Object {
              "large": "27px",
              "medium": "19px",
              "small": "15px",
              "xlarge": "30px",
              "xsmall": "14px",
            },
            "lineHeights": Object {
              "large": "40px",
              "medium": "32px",
              "small": "24px",
              "xlarge": "44px",
              "xsmall": "24px",
            },
          },
          "tertiary": Object {
            "fontFamily": "FiraSans, sans-serif",
            "fontSizes": Object {
              "large": "29px",
              "medium": "21px",
              "small": "17px",
              "xlarge": "35px",
              "xsmall": "15px",
            },
            "lineHeights": Object {
              "large": "40px",
              "medium": "32px",
              "small": "24px",
              "xlarge": "44px",
              "xsmall": "24px",
            },
          },
          "utility": Object {
            "fontFamily": "ProximaNovaCond, sans-serif",
            "fontSizes": Object {
              "large": "32px",
              "medium": "24px",
              "small": "19px",
              "xlarge": "37px",
              "xsmall": "16px",
            },
            "lineHeights": Object {
              "large": "40px",
              "medium": "32px",
              "small": "24px",
              "xlarge": "44px",
              "xsmall": "24px",
            },
          },
        },
      }
    }
  >
    <ThemeProvider
      theme={
        Object {
          "blog": "jalopnik",
          "breakpointBanner": "969px",
          "breakpointMedium": "53.125em",
          "color": Object {
            "alert": "#EEB544",
            "amazon": "#F5C056",
            "backgroundLayer": "rgba(0,0,0,0.45)",
            "backgroundLight": "#FFF3EB",
            "black": "#000",
            "blackOverlay": "rgba(0,0,0,0.7)",
            "bodytext": "#222",
            "commerce": "#72AD75",
            "darkBlackOverlay": "rgba(0,0,0,0.9)",
            "darkgray": "#666",
            "darksmoke": "#222",
            "error": "#D24A1D",
            "facebook": "#1877F2",
            "foregroundLayer": "rgba(0,0,0,0.4)",
            "google": "#DB4437",
            "gray": "#7D7D7D",
            "instagram": "#C32AA3",
            "lightgray": "#E5E5E5",
            "linkedin": "#0077B5",
            "logo": "#FF4D00",
            "midgray": "#CCC",
            "primary": "#3E79B1",
            "primaryLight": "#FF7033",
            "rss": "#7D7D7D",
            "secondarytext": "#7D7D7D",
            "success": "#23891F",
            "twitter": "#1DA1F2",
            "white": "#FFF",
            "whiteOverlay": "rgba(255,255,255,0.7)",
            "whitesmoke": "#F5F5F5",
            "youtube": "#FF0000",
          },
          "columnGutter": "2.25rem",
          "columnGutter19": Object {
            "largeUp": "1.5rem",
            "small19": "1rem",
            "xlargeUp": "2rem",
          },
          "columnPadding": "1.125rem",
          "columnPadding19": Object {
            "largeUp": "0.75rem",
            "small19": "0.5rem",
            "xlargeUp": "1rem",
          },
          "featuredContentWidth": "1024px",
          "globalNavHeight": "80px",
          "linkTransition": "0.1s ease-out",
          "mainContainerWidth": "66.5%",
          "mainContentMaxWidth": "800px",
          "pageWidth": "1336px",
          "postContentMaxWidth": "636px",
          "scrollbackNavHeight": "60px",
          "sidebarContainerWidth": "33.5%",
          "sidebarContentMaxWidth": "360px",
          "sidebarContentMinWidth": "300px",
          "slideshowTransitionDuration": "0.2s",
          "typography": Object {
            "body": Object {
              "fontFamily": "ElizabethSerif, Georgia, serif",
              "fontSizes": Object {
                "large": "27px",
                "medium": "19px",
                "small": "15px",
                "xlarge": "30px",
                "xsmall": "14px",
              },
              "lineHeights": Object {
                "large": "40px",
                "medium": "32px",
                "small": "24px",
                "xlarge": "44px",
                "xsmall": "24px",
              },
            },
            "embiggenedHeadlineSize": "34px",
            "headline": Object {
              "fontFamily": "ProximaNovaCond, sans-serif",
              "fontSizes": Object {
                "large": "32px",
                "medium": "24px",
                "small": "19px",
                "xlarge": "37px",
                "xsmall": "16px",
              },
              "lineHeights": Object {
                "large": "40px",
                "medium": "32px",
                "small": "24px",
                "xlarge": "44px",
                "xsmall": "24px",
              },
            },
            "headlineLineHeight": "1.3",
            "headlineSize": "24px",
            "permalinkHeadlineLineHeight": "1.2",
            "permalinkHeadlineSize": Object {
              "mediumUp": "2.5rem",
              "small": "2rem",
              "xxxlargeUp": "3rem",
            },
            "primary": Object {
              "fontFamily": "ProximaNovaCond, sans-serif",
              "fontSizes": Object {
                "large": "32px",
                "medium": "24px",
                "small": "19px",
                "xlarge": "37px",
                "xsmall": "16px",
              },
              "lineHeights": Object {
                "large": "40px",
                "medium": "32px",
                "small": "24px",
                "xlarge": "44px",
                "xsmall": "24px",
              },
            },
            "serif": Object {
              "fontFamily": "ElizabethSerif, Georgia, serif",
              "fontSizes": Object {
                "large": "27px",
                "medium": "19px",
                "small": "15px",
                "xlarge": "30px",
                "xsmall": "14px",
              },
              "lineHeights": Object {
                "large": "40px",
                "medium": "32px",
                "small": "24px",
                "xlarge": "44px",
                "xsmall": "24px",
              },
            },
            "tertiary": Object {
              "fontFamily": "FiraSans, sans-serif",
              "fontSizes": Object {
                "large": "29px",
                "medium": "21px",
                "small": "17px",
                "xlarge": "35px",
                "xsmall": "15px",
              },
              "lineHeights": Object {
                "large": "40px",
                "medium": "32px",
                "small": "24px",
                "xlarge": "44px",
                "xsmall": "24px",
              },
            },
            "utility": Object {
              "fontFamily": "ProximaNovaCond, sans-serif",
              "fontSizes": Object {
                "large": "32px",
                "medium": "24px",
                "small": "19px",
                "xlarge": "37px",
                "xsmall": "16px",
              },
              "lineHeights": Object {
                "large": "40px",
                "medium": "32px",
                "small": "24px",
                "xlarge": "44px",
                "xsmall": "24px",
              },
            },
          },
        }
      }
    >
      <popular-search__Container
        blogTheme="jalopnik"
      >
        <StyledComponent
          blogTheme="jalopnik"
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "popular-search__Container-uwatm0-3",
                "isStatic": false,
                "lastClassName": "c0",
                "rules": Array [
                  "width:100%;display:flex;flex-direction:column;text-align:center;",
                  ".c1",
                  ",",
                  ".c4",
                  "{font-weight:bold;font-size:15px;line-height:19px;}",
                  "@media only screen and ",
                  "(min-width: 50em)",
                  "{",
                  "
		padding: 8px 10px 10px;
		border: 1px solid ",
                  [Function],
                  ";

		",
                  ".c1",
                  " {
			margin-bottom: 14px;
			font-size: 18px;
			line-height: 22px;
		}

		",
                  ".c4",
                  " {
			font-size: 20px;
			line-height: 24px;
			margin-bottom: 8px;
		}
	",
                  "}",
                ],
              },
              "displayName": "popular-search__Container",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "popular-search__Container-uwatm0-3",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c0"
          >
            <popular-search__Title>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "c1",
                      "isStatic": true,
                      "lastClassName": "c2",
                      "rules": Array [
                        "margin-bottom:10px;",
                      ],
                    },
                    "displayName": "popular-search__Title",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "c1",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <span
                  className="c1 c2"
                >
                  Popular searches on Jalopnik
                </span>
              </StyledComponent>
            </popular-search__Title>
            <popular-search__KeywordContainer>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "popular-search__KeywordContainer-uwatm0-2",
                      "isStatic": true,
                      "lastClassName": "c3",
                      "rules": Array [
                        "display:flex;flex-direction:column;align-items:center;margin:0;",
                      ],
                    },
                    "displayName": "popular-search__KeywordContainer",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "popular-search__KeywordContainer-uwatm0-2",
                    "target": "ul",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <ul
                  className="c3"
                >
                  <popular-search__Keyword
                    key="Uruk-hai"
                    onClick={[Function]}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "c4",
                            "isStatic": false,
                            "lastClassName": "c5",
                            "rules": Array [
                              "margin-bottom:4px;color:",
                              [Function],
                              ";cursor:pointer;&:hover{color:",
                              [Function],
                              ";}",
                            ],
                          },
                          "displayName": "popular-search__Keyword",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "c4",
                          "target": "li",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      onClick={[Function]}
                    >
                      <li
                        className="c4 c5"
                        onClick={[Function]}
                      >
                        Uruk-hai
                      </li>
                    </StyledComponent>
                  </popular-search__Keyword>
                  <popular-search__Keyword
                    key="Goblins"
                    onClick={[Function]}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "c4",
                            "isStatic": false,
                            "lastClassName": "c5",
                            "rules": Array [
                              "margin-bottom:4px;color:",
                              [Function],
                              ";cursor:pointer;&:hover{color:",
                              [Function],
                              ";}",
                            ],
                          },
                          "displayName": "popular-search__Keyword",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "c4",
                          "target": "li",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      onClick={[Function]}
                    >
                      <li
                        className="c4 c5"
                        onClick={[Function]}
                      >
                        Goblins
                      </li>
                    </StyledComponent>
                  </popular-search__Keyword>
                  <popular-search__Keyword
                    key="Moria Orcs"
                    onClick={[Function]}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "c4",
                            "isStatic": false,
                            "lastClassName": "c5",
                            "rules": Array [
                              "margin-bottom:4px;color:",
                              [Function],
                              ";cursor:pointer;&:hover{color:",
                              [Function],
                              ";}",
                            ],
                          },
                          "displayName": "popular-search__Keyword",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "c4",
                          "target": "li",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      onClick={[Function]}
                    >
                      <li
                        className="c4 c5"
                        onClick={[Function]}
                      >
                        Moria Orcs
                      </li>
                    </StyledComponent>
                  </popular-search__Keyword>
                  <popular-search__Keyword
                    key="Black Uruks"
                    onClick={[Function]}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "c4",
                            "isStatic": false,
                            "lastClassName": "c5",
                            "rules": Array [
                              "margin-bottom:4px;color:",
                              [Function],
                              ";cursor:pointer;&:hover{color:",
                              [Function],
                              ";}",
                            ],
                          },
                          "displayName": "popular-search__Keyword",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "c4",
                          "target": "li",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      onClick={[Function]}
                    >
                      <li
                        className="c4 c5"
                        onClick={[Function]}
                      >
                        Black Uruks
                      </li>
                    </StyledComponent>
                  </popular-search__Keyword>
                </ul>
              </StyledComponent>
            </popular-search__KeywordContainer>
          </div>
        </StyledComponent>
      </popular-search__Container>
    </ThemeProvider>
  </Theme>
</PopularSearch>
`;
